gapic-generator/lib/gapic/presenters/resource_presenter.rb

```diff
@@ -52,6 +54,25 @@ def path_helper
         "#{ActiveSupport::Inflector.underscore name}_path"
       end
 
+      ##
+      # Deduplicates pattern that have the same `arguments_key`. Our design for the "paths" helper
+      # only allows for one pattern per `arguments_key`.
+      #
+      # If patterns with the same `arguments_key` are detected, the shortest is taken. If there is
+      # a tie, the lexicographically first is taken.
+      #
+      # @param patterns [Array<PatternPresenter>]
+      #
+      # @return [Array<PatternPresenter>]
+      #
+      def self.dedup_patterns patterns
+        patterns.group_by(&:arguments_key).map do |arguments_key, group|
+          group.sort_by do |pattern|
+            [pattern.pattern.length, pattern.pattern]
+          end.first
```

Comment:
I like the functional Ruby!

Instead of sort_by...first, try min_by.

Reply:
Done